ports/stm32/boards/stm32f4xx_prefix.c

```diff
@@ -30,3 +30,19 @@
         .adc_num = p_adc_num, \
         .adc_channel = p_adc_channel, \
     }
+
+#if MICROPY_HW_ENABLE_ANALOG_ONLY_PINS
+#define PIN_ANALOG(p_port, p_pin, p_af, p_adc_num, p_adc_channel) \
+    { \
+        { &pin_analog_type }, \
+        .name = MP_QSTR_##p_port##p_pin##_C, \
+        .port = PORT_##p_port, \
+        .pin = (p_pin), \
+        .num_af = (sizeof(p_af) / sizeof(pin_af_obj_t)), \
+        .pin_mask = (1 << ((p_pin) & 0x0f)), \
+        .gpio = GPIO##p_port, \
+        .af = p_af, \
+        .adc_num = p_adc_num, \
+        .adc_channel = p_adc_channel, \
+    }
+#endif
```

Comment:
Could we add an `#else` here that did something like

```
#define PIN_ANALOG DUAL_PAD_SUPPORT_NOT_ENABLED__CONFIGURE__MICROPY_HW_ANALOG_SWITCH_PAx
```

Reply:
Done.